backend/src/main/java/meowhub/backend/jpa_buddy/GroupchatMessage.java

```diff
@@ -0,0 +1,80 @@
+package meowhub.backend.jpa_buddy;
+
+import jakarta.persistence.Column;
+import jakarta.persistence.Entity;
+import jakarta.persistence.FetchType;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.JoinColumn;
+import jakarta.persistence.ManyToOne;
+import jakarta.persistence.OneToMany;
+import jakarta.persistence.Table;
+import jakarta.validation.constraints.NotNull;
+import jakarta.validation.constraints.Size;
+import lombok.Getter;
+import lombok.Setter;
+import org.hibernate.annotations.ColumnDefault;
+import org.hibernate.annotations.OnDelete;
+import org.hibernate.annotations.OnDeleteAction;
+
+import java.time.LocalDate;
+import java.util.LinkedHashSet;
+import java.util.Set;
+
+@Getter
+@Setter
+@Entity
+@Table(name = "GROUPCHAT_MESSAGES", schema = "mh_groups")
+public class GroupchatMessage {
+    @Id
+    @Size(max = 36)
+    @ColumnDefault("sys_guid()")
+    @GeneratedValue(strategy = GenerationType.UUID)
+    @Column(name = "ID", nullable = false, length = 36)
+    private String id;
+
+    @NotNull
+    @ManyToOne(fetch = FetchType.LAZY, optional = false)
+    @OnDelete(action = OnDeleteAction.RESTRICT)
+    @JoinColumn(name = "GROUP_ID", nullable = false)
+    private Group group;
+
+    @NotNull
+    @ManyToOne(fetch = FetchType.LAZY, optional = false)
+    @OnDelete(action = OnDeleteAction.RESTRICT)
+    @JoinColumn(name = "USER_ID", nullable = false)
+    private User user;
+
+    @Size(max = 2000)
+    @NotNull
+    @Column(name = "MESSAGE", nullable = false, length = 2000)
+    private String message;
+
+    @ManyToOne(fetch = FetchType.LAZY)
+    @OnDelete(action = OnDeleteAction.RESTRICT)
+    @JoinColumn(name = "ANSWERED_MESSAGE_ID")
+    private GroupchatMessage answeredMessage;
+
+    @NotNull
+    @Column(name = "MSG_DATE", nullable = false)
+    private LocalDate msgDate;
+
+    @Column(name = "CREATED_AT")
```

Comment:
you may have noticed, but i deleted @NotNull and nullable=false annotation for createdBy and createdAt.
These fields are populated by an audit trigger that is not visible from hibernate's perspective.
@NotNull was throwing an error, because it checks on application level before save, if the field is not null.
nullable = false was deleted, because when running tests for h2, (h2 database for tests is populated by hibernate with models) was throwing an exception, because the triggers are not created there. For production environment is still works correctly, as we are using database-first approach

Reply:
Fair enough 😊